src/jdns/jdns.c

```diff
@@ -2067,6 +2067,10 @@ int _unicast_do_writes(jdns_session_t *s, int now)
 		}
 
 		// in theory, it is not possible for 'ns' to be null here
+		if (ns == 0)
+		{
+			continue; // but lets make sure it will not stab us in the back
```

Comment:
The failure chain is pretty long, I hope the attached trace helps.
![image](https://cloud.githubusercontent.com/assets/1467368/24817082/67d2720a-1bdb-11e7-9ba5-539b17874603.png)